camera/camera_ndk.c

```diff
@@ -143,6 +153,18 @@ int openCamera(int index, int width, int height) {
 }
 
 int captureCamera() {
+    pthread_mutex_lock(&cameraState.mutex);
+    if (cameraState.status == CAMERA_ACTIVE) {
+        LOGW("camera is already active.\n");
+        pthread_mutex_unlock(&cameraState.mutex);
+        return ACAMERA_OK;
```

Comment:
I return OK when active, which does not prevent the `VideoReaderFunc` from returning the same image more than once. If we want guarantees that each request returns a unique frame then we would need to add additional logic to wait for the requests to flush. In that case, it may be more optimal to have a separate routine for captureRequest handling and a nonce to ensure the latest frame returned is newer.